pkg/rbac/roles.go

```diff
@@ -0,0 +1,75 @@
+package rbac
+
+import (
+	"fmt"
+	"github.com/tektoncd/triggers/pkg/apis/triggers/v1alpha1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	// RolePostfix is the postfix added to the name of the role that will be created to review trigger resources
+	RolePostfix string = "-trigger-role"
+	// RoleBindingPostfix is the postfix added to the name of the role binding that will be bound to the EventListener service account
+	RoleBindingPostfix string = "-trigger-rolebinding"
+)
+
+// AddRoleBinding creates a role and role binding in the EventListener namespace for the EventListener's ServiceAccount to view Triggers resources.
+// Both the role and role binding have owner references on the EventListener.
+// This function is idempotent since it should be utilized by the EventListener reconciler.
+func AddRoleBinding(eventListener *v1alpha1.EventListener, c kubernetes.Interface) error {
+	role := &rbacv1.Role{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            fmt.Sprintf("%s%s", eventListener.Spec.ServiceAccountName, RolePostfix),
+			Namespace:       eventListener.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*eventListener.GetOwnerReference()},
+		},
+		Rules: []rbacv1.PolicyRule{
+			rbacv1.PolicyRule{
+				APIGroups: []string{
+					"tekton.dev",
+				},
+				Resources: []string{
+					"eventlisteners",
+					"triggerbindings",
+					"triggertemplates",
+				},
+				Verbs: []string{
+					"get",
+				},
+			},
+		},
+	}
+	if _, err := c.RbacV1().Roles(eventListener.Namespace).Create(role); err != nil {
+		if !errors.IsAlreadyExists(err) {
+			return err
+		}
+	}
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:            fmt.Sprintf("%s%s", eventListener.Spec.ServiceAccountName, RoleBindingPostfix),
+			Namespace:       eventListener.Namespace,
+			OwnerReferences: []metav1.OwnerReference{*eventListener.GetOwnerReference()},
+		},
+		Subjects: []rbacv1.Subject{
+			rbacv1.Subject{
+				Kind:      "ServiceAccount",
+				Name:      eventListener.Spec.ServiceAccountName,
+				Namespace: eventListener.Namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     fmt.Sprintf("%s%s", eventListener.Spec.ServiceAccountName, RolePostfix),
```

Comment:
nit: `fmt.Sprintf("%s%s", eventListener.Spec.ServiceAccountName, RolePostfix)` is repeated from [here](https://github.com/tektoncd/triggers/pull/73/files#diff-ddd5c0a5fb9951edd4647bf70a8e9453R25). Might be nice to store this as a variable somewhere for reuse.